src/client/Microsoft.Identity.Client/Kerberos/KerberosSupplementalTicketManager.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Json.Linq;
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.Identity.Client.Kerberos
+{
+    /// <summary>
+    /// Helper class to manage Kerberos Ticket Claims.
+    /// </summary>
+    public static class KerberosSupplementalTicketManager
+    {
+        private const int DefaultLogonId = 0;
+        private const string KerberosClaimType = "xms_as_rep";
+        private const string IdTokenAsRepTemplate = @"{{""id_token"": {{ ""xms_as_rep"":{{""essential"":""false"",""value"":""{0}""}} }} }}";
+        private const string AccessTokenAsRepTemplate = @"{{""access_token"": {{ ""xms_as_rep"":{{""essential"":""false"",""value"":""{0}""}} }} }}";
+
+        /// <summary>
+        /// Creates a <see cref="KerberosSupplementalTicket"/> object from given ID token string..
+        /// </summary>
+        /// <param name="idToken">ID token string.</param>
+        /// <returns>A <see cref="KerberosSupplementalTicket"/> object if exists and parsed correctly. Null, otherwise.</returns>

```

Comment:
```suggestion
        /// <returns>A <see cref="KerberosSupplementalTicket"/> object if exists in the <see cref="idToken"/> and is parsed correctly. Null, otherwise.</returns>

```

Reply:
Updated.